R/emissions_profile_any_compute_profile_ranking.R

```diff
@@ -25,12 +25,19 @@
 emissions_profile_any_compute_profile_ranking <- function(data) {
   check_emissions_profile_any_compute_profile_ranking(data)
 
-  if (hasName(data, "profile_ranking")) {
-    out <- check_crucial_names(data, "grouped_by")
-  } else {
-    benchmarks <- set_names(epa_benchmarks(data), flat_benchmarks(data))
-    out <- map_df(benchmarks, ~ add_rank(data, .x), .id = "grouped_by")
-  }
+  exclude <- short_isic(data) |
+    is.na(get_column(data, aka("isic"))) |
+    is.na(get_column(data, aka("tsector")))
+
+  list(!exclude, exclude) |>
+    map(\(x) filter(data, x)) |>
+    map_df(\(data) emissions_profile_any_compute_profile_ranking_impl(data)) |>
+    assign_na_to_profile_ranking_in_special_cases()
```

Comment:
@kalashsinghal 

Here you can see that we first exclude the special cases and only then we compute `profile_ranking`. I also calculate `profile_ranking` for the rows we want to `exclude`  but later I convert them to `NA`. This seems silly but it allows us to extend rather than modify the original code -- following the Open-close (SOLID) principle that "code should be open to extension but closed to modification".

Reply:
@maurolepore This was a very new logic for me to understand, however its clear now. I agree to the Open-close principle. LGTM

Reply:
And yes I want this change for all the special cases which you did correctly. Please feel free to merge